lib/pbench/agent/results.py

```diff
@@ -406,8 +406,13 @@ def __init__(
         if server:
             path = config.get("results", "rest_endpoint")
             uri = f"{server}/{path}"
+            self.ca = True
         else:
             uri = config.get("results", "server_rest_url")
+
+            # If the "server_ca" config variable isn't defined, we expect to verify
+            # using a registered CA or via REQUESTS_CA_BUNDLE environment variable.
+            self.ca = config.get("results", "server_ca", fallback=True)
```

Comment:
Yeah, I'm inclined to use `--server` and see if we can escape using a custom `pbench-agent.cfg` file in the typical case.

Reply:
I think I'm content for now (especially since I removed the default `pbench-agent.cfg` mapping from the script) to deal with two specific cases:

1. relies on `PB_AGENT_CA=<file>` to map the CA file and set `REQUESTS_CA_BUNDLE` inside the container, and uses `pbench-results-move --server`, which won't take the code path that even looks for `[results] server_ca`; the `--server` path sets `self.ca = None`, which is what we want.
2. we're using a customized `pbench-agent.cfg` file with the proper server address *and* `[results] server_ca` where necessary; if the CA isn't specified, `verify=True` does what we want, (effectively the default), forcing validation against the standard `certifi` chain.

Effectively, whether there'd actually be a problem with `REQUESTS_CA_BUNDLE` and `verify=True` becomes irrelevant.

Technically, though, `verify=True` has the same effect as the default `verify=None`, so I might as well drop the fallback even though it shouldn't matter a bit.

I've tested with

```
PB_AGENT_PODMAN_OPTIONS="-v $(realpath server/pbenchinacan/etc/pki/tls/certs/pbench_CA.crt):/etc/pki/tls/certs/pbench_CA.crt:z -v $(realpath ~/.config/pbench/pbench-agent.cfg):/opt/pbench-agent/config/pbench-agent.cfg:z" \
    ./contrib/containerized-pbench/pbench pbench-results-move --token ${TOKEN}
```

with

`~/.config/pbench/pbench-agent.cfg`:
```
[DEFAULT]
pbench_install_dir = /opt/pbench-agent
pbench_web_server = dhcp-231.ctrl.perf-infra.lab.eng.rdu2.redhat.com:8443
[config]
path = %(pbench_install_dir)s/config
files = pbench-agent-default.cfg
[results]
server_ca = /etc/pki/tls/certs/pbench_CA.crt
```

and the simpler `--server` case as:

```
PB_AGENT_CA=$(realpath server/pbenchinacan/etc/pki/tls/certs/pbench_CA.crt) \ 
    ./contrib/containerized-pbench/pbench pbench-results-move --token ${TOKEN} --server https://dhcp-231.ctrl.perf-infra.lab.eng.rdu2.redhat.com:8443
```

Both work.

Note that to test all this with a few debugging hacks along the way, I had to locate the "tricks and tips" we worked out for doing a local Agent container build: I put that into a new script to make it easy to find.